cfme/storage/manager.py

```diff
@@ -0,0 +1,233 @@
+from cfme.utils.appliance import BaseCollection, BaseEntity
+
+
+class ManagerToolbar(View):
+    """The toolbar on the Storage Manager or Provider page"""
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+
+
+class ManagerDetailsToolbar(View):
+    """The toolbar on the Storage Manager or Provider detail page"""
+    reload = Button(title='Reload Current Display')
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    monitoring = Dropdown('Monitoring')
+    download = Button(title='Download summary in PDF format')
+
+
+class ManagerEntities(View):
+    """The entities on the main list Storage Manager or Provider page"""
+    table = Table(".//div[@id='list_grid']/table")
+
+
+class ManagerDetailsEntities(View):
+    """The entities on the Storage Manager or Provider details page"""
+    breadcrumb = BreadCrumb()
+    properties = SummaryTable('Properties')
+    relationships = SummaryTable('Relationships')
+    smart_management = SummaryTable('Smart Management')
+    status = SummaryTable('Status')
+
+
+class ManagerDetailsAccordion(View):
+    """The accordion on the Storage Manager or Provider details page"""
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class ManagerView(BaseLoggedInPage):
+    """A base view for all the Storage Manager or Provider pages"""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+
+    flash = FlashMessages(
+        './/div[@id="flash_msg_div"]/div[@id="flash_text_div" or '
+        'contains(@class, "flash_text_div")]')
+
+    @property
+    def in_manager(self):
+        nav_path = self.context['object'].nav.pick(self.context['object'].appliance.version)
+        return(
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == nav_path)
+
+
+class ManagerAllView(ManagerView):
+    """The all Storage Manager or Provider page"""
+    @property
+    def is_displayed(self):
+        return (
+            self.in_manager and
+            self.title.text in ('Storage Managers', self.context['object'].manager_type))
+
+    toolbar = View.nested(ManagerToolbar)
+    entities = View.nested(ManagerEntities)
+    paginator = PaginationPane()
+
+
+class ManagerDetailsView(ManagerView):
+    """The details page for Storage Manager or Provider"""
+    @property
+    def is_displayed(self):
+        expected_title = '{} (Summary)'.format(self.context['object'].name)
+
+        return(
+            self.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+    toolbar = View.nested(ManagerDetailsToolbar)
+    sidebar = View.nested(ManagerDetailsAccordion)
+    entities = View.nested(ManagerDetailsEntities)
+
+
+class ManagerTagsView(ManagerView):
```

Comment:
@psav , @rlbabyuk, tags/policies views are the same everywhere. Do we have any such shared views to re-use here ?

Reply:
@izapolsk   Yup you are right, I referred @rlbabyuk  PR about TagPageView and WidgetasticTaggable. removing it.